app/interactions/assignment_interactions/update_in_progress.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# When an Assignment is marked as in progress, we need to check to see if the
+# assignment was in a state where the reward had already been applied and, if so,
+# roll that back.
+# Rescue any errors, and always log what we did here.
+# TODO: Eventually we will also want to rollback any consequences that may have
+#       been applied as a result of this Assignment having been marked failed.
+module AssignmentInteractions
+  # Class to handle logic after an Assignment has been updated to :in_progress
+  class UpdateInProgress < BaseAssignmentInteraction
+    def initialize(options)
+      super
+    end
+
+    # This should be the only public method in the interaction
+    def run
+      if @assignment.reward_applied?
```

Comment:
This seems unnecessary? Maybe reward rollback in general is unnecessary because rewards should never actually be applied until an assignment is marked complete, at which point it's final? idk, food for thought.